xarray/core/merge.py

```diff
@@ -1,173 +1,368 @@
-from .alignment import align, partial_align, align_variables
-from .utils import ChainMap
-from .variable import as_variable
+from collections import defaultdict, deque
+
+import pandas as pd
+
+from .alignment import align, deep_align
+from .utils import ChainMap, Frozen
+from .variable import as_variable, Coordinate, Variable, default_index_coordinate
 from .pycompat import (basestring, iteritems, OrderedDict)
 
 
-def _as_dataset_variable(name, var):
-    """Prepare a variable for adding it to a Dataset
+def broadcast_dimension_size(variables):
+    # type: (List[Variable],) -> Variable
```

Comment:
out of curiosity, what's with the type comment at the start of your new functions?